server/src/test/java/org/apache/kafka/server/ClientMetricsManagerTest.java

```diff
@@ -919,4 +922,101 @@ public void testPushTelemetryConcurrentRequestAfterSubscriptionUpdate() throws U
         // 1 request should fail with throttling error.
         assertEquals(1, throttlingErrorCount);
     }
+
+    @Test
+    public void testCacheEviction() throws UnknownHostException, InterruptedException {
+        Properties properties = new Properties();
+        properties.put("metrics", ClientMetricsConfigs.ALL_SUBSCRIBED_METRICS_CONFIG);
+        properties.put(ClientMetricsConfigs.PUSH_INTERVAL_MS, "100");
+        clientMetricsManager.updateSubscription("sub-1", properties);
+
+        GetTelemetrySubscriptionsRequest request = new GetTelemetrySubscriptionsRequest.Builder(
+            new GetTelemetrySubscriptionsRequestData(), true).build();
+
+        GetTelemetrySubscriptionsResponse response = clientMetricsManager.processGetTelemetrySubscriptionRequest(
+            request, ClientMetricsTestUtils.requestContext());
+        assertEquals(Errors.NONE, response.error());
+
+        assertNotNull(clientMetricsManager.clientInstance(response.data().clientInstanceId()));
+        assertEquals(1, clientMetricsManager.expirationTimer().size());
+        // Advance the clock to trigger cache eviction, cache expiry should be 100 * 3 = 300 ms.
+        clientMetricsManager.expirationTimer().advanceClock(300);
+        assertTimeoutPreemptively(Duration.ofMillis(300), () -> {
+            // Validate that cache eviction happens and client instance is removed from cache.
+            while (clientMetricsManager.expirationTimer().size() != 0 || clientMetricsManager.clientInstance(response.data().clientInstanceId()) != null) {
```

Comment:
nit: Break this super-long line please, just like the equivalent one in the next test.

Reply:
Done.